File: Codechef/alkh_iiita.cpp
#include<iostream>
#include<algorithm>
#include<cstdio>
using namespace std;
long int sequence(long int numbers[],long int size)
{
      long int max_so_far  = numbers[0], max_ending_here = numbers[0];
        for(long int i = 1; i < size; i++)
        {
                if(max_ending_here > 0)
                {
                        max_ending_here = numbers[i];
                }
                else
                {
                        max_ending_here += numbers[i];
                }
                if(max_ending_here <= max_so_far )
                {
                        max_so_far  = max_ending_here;
                }
        }
        return max_so_far ;
}
int main(){
int t;
scanf("%d",&t);
while(t--){
long int n;
scanf("%ld",&n);
long int a[n];
long int sum = 0;
for(long int i = 0 ; i < n;i++){
scanf("%ld",&a[i]);
sum += a[i];
}
long int ss = sequence(a,n);
if(ss<0){
ss*=-1;
printf("%ld\n",sum+2*ss);}
else{
printf("%ld\n",sum);}
}
return 0;
}

File: Codechef/alkh_iiita.py
int sequence(std::vector<int> const & numbers)
{
        // Initialize variables here
        int max_so_far  = numbers[0], max_ending_here = numbers[0];
 
        // OPTIONAL: These variables can be added in to track the position of the subarray
        // size_t begin = 0;
        // size_t begin_temp = 0;
        // size_t end = 0;
 
        // Find sequence by looping through
        for(size_t i = 1; i < numbers.size(); i++)
        {
                // calculate max_ending_here
                if(max_ending_here < 0)
                {
                        max_ending_here = numbers[i];
 
                        // begin_temp = i+1;
                }
                else
                {
                        max_ending_here += numbers[i];
                }
 
                // calculate max_so_far
                if(max_ending_here >= max_so_far )
                {
                        max_so_far  = max_ending_here;
 
                        // begin = begin_temp;
                        // end = i;
                }
        }
        return max_so_far ;
}
